tests/testthat/test-generate_data.R

```diff
@@ -280,3 +280,38 @@ test_that("genFactor works.", {
   expect_length(genFactor(copy(dt), c("q1", "q2"), replace = TRUE), 4)
   expect_equal(genFactor(copy(dt), c("q1", "q2"), labels = labels), dt_res)
 })
+
+#genDummy ----
+test_that("genDummy throws errors.", {
+  oldSeed <- .Random.seed
+  set.seed(10076)
+  
+  d1 <- defData(varname = "rx", formula = "1;1", dist = "trtAssign")
+  d1 <- defData(d1, varname = "male", formula = .4, dist = "binary")
+  d1 <- defData(d1, varname = "z", formula = "0 - 1.2*rx - 1*male", dist = "nonrandom")
+  
+  dd <- genData(5, d1)
+  
+  # Initial data checks
+  expect_error(genDummy(varname = "rx", sep = ".", replace = FALSE))
+  expect_error(genDummy(dd, sep = ".", replace = FALSE))
+  
+  # Check if data table exists
+  expect_error(genDummy(dt, varname = "rx", sep = ".", replace = FALSE))
+  
+  # Check if varname exists
+  expect_error(genDummy(dd, varname = "xx", sep = ".", replace = FALSE))
+  
+  #I do not understand how to test for is.integer or is.factor, I'm not clear on the 
```

Comment:
In addition to what Keith already said, the best way to test for them is [expect_type](https://testthat.r-lib.org/reference/inheritance-expectations.html).

Reply:
Also the [adv-r book](https://adv-r.hadley.nz/vectors-chap.html#factors) has some info on this, overall it and [r-pkgs](https://r-pkgs.org/) are great resources :)